### PR TITLE
Fixes #20914 - Connect to configured Foreman DB during export

### PR DESCRIPTION
**Pull request by mbacovsky**
foreman-debug extension was updated to use DB from Foreman configuration
to export foreman-tasks DB data. Before the patch connection to local
postgresql server and database 'foreman' was hard-coded.

### CODE REVIEW

File: extra/dynflow-debug.sh [hunk 1]
```diff
@@ -3,8 +3,11 @@
 # This file provides additional debug information for foreman-debug tool and is
 # symlinked as /usr/share/foreman/script/foreman-debug.d/60-dynflow_debug
 
+GET_DB_CONFIG_SCRIPT='db = YAML.load_file("/etc/foreman/database.yml")["production"];puts %(PGPASSWORD="#{db["password"]}" psql -U #{db["username"]} -h "#{db["host"]||"localhost"}" -p #{db["port"]||"5432"} -d #{db["database"]})'
```
Comment (adamruzicka) [inline extra/dynflow-debug.sh]: Whoa, this is pretty wild. Any reason for not having it span multiple lines?
```shell
FOREMAN_PSQL_COMMAND=$(ruby -ryaml <<-END
db = YAML.load_file("/etc/foreman/database.yml")["production"]
puts [
  "PGPASSWORD='#{db['password']}'",
  'psql',
  '-U', "'#{db.fetch('username')}'",
  '-h', "'#{db['host'] || 'localhost'}'",
  '-p', "'#{(db['port'] || 5432).to_s}'",
  '-d', "'#{db.fetch('database')}'"
].join(' ')
END
)
```

**Comment by mbacovsky** [comment]
@adamruzicka, good point, I was struggling with the right quoting. Thanks for the suggestion

**Comment by adamruzicka** [comment]
As per discussion on IRC, we should check for database type and do the export only if the db is postgresql and skip it for any other.

**Comment by mbacovsky** [comment]
@adamruzicka updated

**Comment by adamruzicka** [comment]
Works as expected. @iNecas do we really need the tests to be green for this?

**Comment by iNecas** [comment]
We don't

**Comment by iNecas** [comment]
Thanks @mbacovsky and @adamruzicka

**Pull request merged**